### ISSUE TITLE
`code-block-tools`: use native `--check` commands for explicit format-check variants

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#900]
## Summary

As a follow-up to #888 and:
 - **code-block-tools**: add explicit rumdl, oxfmt, djlint, and shuck modes (https://github.com/rvben/rumdl/commit/77f0e55f1d5d53ad65f0fd6f4688a09f9a82bf7d)

Please update the explicit format-check variants that currently emulate checking by running a formatter and comparing its output.

The other variants added by #888 already select the intended lint, fix, or format operation and are out of scope.

## Proposed commands

| Tool ID | Current behavior | Expected command |
| --- | --- | --- |
| `oxfmt:lint` | Run the formatter and compare its output | `oxfmt --check` |
| `djlint:html:format-check` | Run `djlint --reformat --profile=html` and compare its output | `djlint --check` |
| `djlint:jinja:format-check` | Run `djlint --reformat --profile=jinja` and compare its output | `djlint --check --profile=jinja` |
| `shuck:format-check` | Run `shuck format` and compare its output | `shuck format --check` |

Required stdin-specific arguments such as `-`, `--stdin-filepath`, or profile/output arguments may still be added where necessary for Code Block Tools, but each variant should use the tool's native check mode rather than emulate it through formatting.

A native check result indicating that formatting changes are required should be reported as a Code Block Tools finding, not as a generic tool execution failure.

## Usage

```toml
[code-block-tools]
enabled = true

[code-block-tools.languages]
javascript = { lint = ["oxfmt:lint"] }
html = { lint = ["djlint:html:format-check"] }
jinja = { lint = ["djlint:jinja:format-check"] }
shell = { lint = ["shuck:format-check"] }
```

Run:

```console
$ rumdl check docs/
```

Each configured variant should invoke its native check mode and report unformatted code blocks without modifying them.

## References

- https://github.com/rvben/rumdl/blob/main/docs/code-block-tools.md
- #888
- https://github.com/rvben/rumdl/commit/77f0e55f1d5d53ad65f0fd6f4688a09f9a82bf7d
